framework/src/timesteppers/FunctionDT.C

```diff
@@ -41,12 +41,20 @@ FunctionDT::FunctionDT(const InputParameters & parameters)
   : TimeStepper(parameters),
     _time_t(getParam<std::vector<Real>>("time_t")),
     _time_dt(getParam<std::vector<Real>>("time_dt")),
-    _time_ipol(_time_t, getParam<std::vector<Real>>("time_dt")),
     _growth_factor(getParam<Real>("growth_factor")),
     _cutback_occurred(false),
     _min_dt(getParam<Real>("min_dt")),
     _interpolate(getParam<bool>("interpolate"))
 {
+  try
+  {
+    _time_ipol = libmesh_make_unique<LinearInterpolation>(_time_t, _time_dt);
+  }
+  catch (std::domain_error & e)
```

Comment:
Side issue, I found it interesting that we throw two different types of std::exceptions (std::out_of_range and std::domain_error) from the `LinearInterpolation` but based on the [docs](http://en.cppreference.com/w/cpp/error/domain_error) it would seem that they are all actually domain errors.